core/src/tile/tileManager.cpp

```diff
@@ -583,19 +586,18 @@ bool TileManager::updateProxyTile(TileSet& _tileSet, TileEntry& _tile,
     {
         const auto& it = tiles.find(_proxyTileId);
         if (it != tiles.end()) {
-            auto& entry = it->second;
-
-            if (!entry.isCanceled() && _tile.setProxy(_proxyId)) {
+            if (_tile.setProxy(_proxyId)) {
+                auto& entry = it->second;
                 entry.incProxyCounter();
 
                 if (entry.isReady()) {
                     m_tiles.push_back(entry.tile);
                 }
-
-                // Note: No need to check the cache: When the tile is in
-                // tileSet it would have already been fetched from cache
                 return true;
             }
+            // Note: No need to check the cache: When the tile is in
```

Comment:
Not necessary now! (Also rename "tile" to "tileEntry" here!)

A tileEntry could be sitting in the tileSet even without any task/tile, as per the change here: https://github.com/tangrams/tangram-es/blob/4de407f92284649cadc4def9c5544c4a0ece0cb8/core/src/tile/tileManager.cpp#L413-L420

Though we still should have the early return here, but update the comment.